numpy/core/src/multiarray/compiled_base.c

```diff
@@ -1210,7 +1210,27 @@ arr_unravel_index(PyObject *self, PyObject *args, PyObject *kwds)
     int i, ret_ndim;
     npy_intp ret_dims[NPY_MAXDIMS], ret_strides[NPY_MAXDIMS];
 
-    char *kwlist[] = {"indices", "dims", "order", NULL};
+    char *kwlist[] = {"indices", "shape", "order", NULL};
+
+    /* Continue to support the older "dims" argument in place
+     * of the "shape" argument. Issue an appropriate warning
+     * if "dims" is detected in keywords, then replace it with
+     * the new "shape" argument and continue processing as usual */
+
+
+    if (kwds) {
```

Comment:
This is not so bad for a function for which speed is probably not of the essence, but wouldn't you rather look for `dims` only if the parsing failed with the newer order?

Reply:
Is it unambiguously clear that there's a substantially faster alternative implementation that both works and maintains code clarity? Obviously, I'll switch if there is.

The primary reasoning behind my strategy here was to avoid touching the presumably solid keyword Error handling code to the extent possible & just swap the new keyword in at the C API level if the old one is used.

Reply:
It is not a big deal. My main logic was not to hinder the "standard" code path at all.